src/util/make-knex.js

```diff
@@ -8,9 +8,8 @@ import QueryInterface from '../query/methods';
 import * as helpers from '../helpers';
 import { assign } from 'lodash'
 import BatchInsert from './batchInsert';
-import Knex from "../index";
 
-export default function makeKnex(client) {
+export default function makeKnex(client, version) {
```

Comment:
Agreed.

Though if it were up to me, I'd add a `version.js` file as an interface layer, have it import the `package.json` file and export just the version information, and then require that `version.js` file from wherever the version is needed. But that's just because importing a '.json' as anything by a localized implementation detail smells bad to me. :imp: